File: VB/PivotGridDrillDown.Module/BusinessObjects/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace PivotGridDrillDown.Module.BusinessObjects
	<DefaultClassOptions> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
		Private _Value1 As Integer
		Public Property Value1() As Integer
			Get
				Return _Value1
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Value1", _Value1, value)
			End Set
		End Property
		Private _Value2 As Integer
		Public Property Value2() As Integer
			Get
				Return _Value2
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Value2", _Value2, value)
			End Set
		End Property
		Private _Value3 As Integer
		Public Property Value3() As Integer
			Get
				Return _Value3
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Value3", _Value3, value)
			End Set
		End Property
	End Class

End Namespace

File: CS/PivotGridDrillDown.Module/BusinessObjects/DomainObject1.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace PivotGridDrillDown.Module.BusinessObjects {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session)
            : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private int _Value1;
        public int Value1 {
            get { return _Value1; }
            set { SetPropertyValue("Value1", ref _Value1, value); }
        }
        private int _Value2;
        public int Value2 {
            get { return _Value2; }
            set { SetPropertyValue("Value2", ref _Value2, value); }
        }
        private int _Value3;
        public int Value3 {
            get { return _Value3; }
            set { SetPropertyValue("Value3", ref _Value3, value); }
        }
    }

}
